src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -302,6 +344,32 @@ def generate_zero_coverage_report(self, report):
         with open('code-coverage-reports/zero_coverage_files.json', 'w') as f:
             json.dump(zero_coverage_files, f)
 
+    def generate_files_list(self, covered=True, chunk=None):
+        options = ['--filter-covered', '--threads', '2']
+        files = self.generate_info(chunk=chunk, out_format='files', options=options)
+        return files.splitlines()
+
+    def generate_chunk_mapping(self):
+        def get_files_task(chunk):
+            return lambda: (chunk, self.generate_files_list(True, chunk=chunk))
+
+        with ThreadPoolExecutor(max_workers=4) as executor:
+            futures = []
+            for chunk in self.get_chunks():
+                futures.append(executor.submit(get_files_task(chunk)))
+
+            with sqlite3.connect('chunk_mapping.db') as conn:
+                c = conn.cursor()
+                c.execute('CREATE TABLE files (path text, chunk text)')
+
+                for future in concurrent.futures.as_completed(futures):
+                    (chunk, files) = future.result()
+                    c.executemany('INSERT INTO files VALUES (?,?)', [(f, chunk) for f in files])
+
+        tar = tarfile.open('code-coverage-reports/chunk_mapping.tar.xz', 'w:xz')
```

Comment:
No `with` here? 😄

Reply:
Strangely, the TarFile class doesn't support compression, but tarfile.open does 😃